spectator-web-spring/src/test/java/com/netflix/spectator/controllers/filter/PrototypeMeasurementFilterTest.java

```diff
@@ -0,0 +1,334 @@
+
+        List<PrototypeMeasurementFilterSpecification.TagFilterSpecification> tagsAyBx = Arrays.asList(
+            new PrototypeMeasurementFilterSpecification.TagFilterSpecification("tagA", "Y"),
+            new PrototypeMeasurementFilterSpecification.TagFilterSpecification("tagB", "X"));
+
+        List<PrototypeMeasurementFilterSpecification.TagFilterSpecification> tagsAzBy = Arrays.asList(
+            new PrototypeMeasurementFilterSpecification.TagFilterSpecification("tagA", "Z"),
+            new PrototypeMeasurementFilterSpecification.TagFilterSpecification("tagB", "Y"));
+
+        valueSpecAxBy = new PrototypeMeasurementFilterSpecification.ValueFilterSpecification();
+        valueSpecAxBy.getTags().addAll(tagsAxBy);
+
+        valueSpecAyBx = new PrototypeMeasurementFilterSpecification.ValueFilterSpecification();
+        valueSpecAyBx.getTags().addAll(tagsAyBx);
+
+        valueSpecAzBy = new PrototypeMeasurementFilterSpecification.ValueFilterSpecification();
+        valueSpecAzBy.getTags().addAll(tagsAzBy);
+
+        meterSpecA = new PrototypeMeasurementFilterSpecification.MeterFilterSpecification(
+            Arrays.asList(valueSpecAxBy));
+        meterSpecB = new PrototypeMeasurementFilterSpecification.MeterFilterSpecification(
+            Arrays.asList(valueSpecAyBx));
+        meterSpecC = new PrototypeMeasurementFilterSpecification.MeterFilterSpecification(
+            Arrays.asList(valueSpecAzBy));
+        meterSpecD = new PrototypeMeasurementFilterSpecification.MeterFilterSpecification(
+            Arrays.asList(valueSpecAxBy, valueSpecAyBx));
+    }
+
+    @Test
+    public void testPatternFromSpec() {
+        List<PrototypeMeasurementFilter.TagFilterPattern> tagPatterns = Arrays.asList(
+            new PrototypeMeasurementFilter.TagFilterPattern(Pattern.compile("tagA"), Pattern.compile("X")),
+            new PrototypeMeasurementFilter.TagFilterPattern(Pattern.compile("tagB"), Pattern.compile("Y")));
+        PrototypeMeasurementFilter.MeterFilterPattern meterPattern
+            = new PrototypeMeasurementFilter.MeterFilterPattern("meterA", meterSpecA);
+
+        Assert.assertEquals(meterPattern.getValues().size(), 1);
+        Assert.assertEquals(meterPattern.getValues().get(0).getTags(), tagPatterns);
+    }
+
+    @Test
+    public void testMetricToPatterns() {
+        PrototypeMeasurementFilterSpecification spec = new PrototypeMeasurementFilterSpecification();
+        spec.getInclude().put("meterA", meterSpecA);
+        spec.getInclude().put(".+B", meterSpecB);
+        spec.getInclude().put(".+C.*", meterSpecC);
+        PrototypeMeasurementFilter filter = new PrototypeMeasurementFilter(spec);
+
+        PrototypeMeasurementFilter.MeterFilterPattern meterPatternA
+            = new PrototypeMeasurementFilter.MeterFilterPattern("meterA", meterSpecA);
+        PrototypeMeasurementFilter.MeterFilterPattern meterPatternB
```

Comment:
Doesn't look like this is getting used. The middle pattern in the assert below is using an empty list.